src/representation_theory/monoid_algebra_basis.lean

```diff
@@ -0,0 +1,189 @@
+/-
+Copyright (c) 2022 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import representation_theory.Rep
+import representation_theory.basic
+
+/-!
+# The structure of the `k[G]`-module `k[Gⁿ]`
+
+This file contains API for the `k[G]`-module `k[Gⁿ]`, where `k` is a commutative ring, `G` is
+a group, and we express the module structure as the representation `G →* End(k[Gⁿ])` induced by the
+diagonal action of `G` on `Gⁿ.`
+
+In particular, we define morphisms of `k`-linear `G`-representations between `k[Gⁿ⁺¹]` and
+`k[G] ⊗ₖ k[Gⁿ]` (on which `G` acts by `ρ(g₁)(g₂ ⊗ x) = (g₁ * g₂) ⊗ x`) which we will later show
+define an isomorphism. From this we will deduce that `k[Gⁿ⁺¹]` is free over `k[G]`.
+The freeness of these modules means we can use them to build a projective resolution of the
+(trivial) `k[G]`-module `k`, which is useful in group cohomology.
+
+## Main definitions
+
+  * representation.of_mul_action
+  * Rep.of_mul_action
+  * to_tensor
+  * of_tensor
+
+## Implementation notes
+
+We express `k[G]`-module structures on a group `V` as `k`-linear representations of `G` on `V`, as
+this simplifies some proofs and also makes it easier to work with different structures on the same
+`V`. This is because `module` is a class, and there is only one notation for scalar multiplication,
+`•`; representations, meanwhile, give us more flexibility, by avoiding typeclass inference.
+
+We bundle the type `k[Gⁿ]` - or more generally `k[H]` when `H` has `G`-action - and the
+representation together as a term of type `Rep k G`, and we call it `Rep.of_mul_action k G H.` This
+is so we can talk about morphisms of representations.
+
+-/
+
+noncomputable theory
+universes v u
+variables (k G : Type u) [comm_ring k]
+
+open monoid_algebra
+
+namespace representation
```

Comment:
I think this section inside `namespace representation` would fit better in `representation_theory/basic.lean`.

Reply:
Thanks, I've moved it - put it in a section called `mul_action`, although maybe it should be called `monoid`